File: Packages/CodeText/Tests/CodeTextTests/Samples/sample.ts
// A small shop's till, for the colour samples (041).
import { EventEmitter } from "node:events";

export interface Receipt {
  readonly item: string;
  readonly total: number;
  readonly at: Date;
}

type Listener = (receipt: Receipt) => void;

enum Currency {
  Pound = "GBP",
  Euro = "EUR",
}

const FLOAT: number = 150.0;
const MASK = 0xff as const;

/** Takes money and remembers it. */
export class Till<T extends Receipt = Receipt> extends EventEmitter {
  private receipts: T[] = [];
  public currency: Currency = Currency.Pound;

  constructor(private float: number = FLOAT) {
    super();
  }

  sell(item: string, price: number, count = 1): Receipt {
    if (count <= 0) {
      throw new RangeError(`cannot sell ${count} of ${item}`);
    }
    const receipt: Receipt = { item, total: price * count, at: new Date() };
    this.receipts.push(receipt as T);
    this.float += receipt.total;
    return receipt;
  }

  get takings(): number {
    return this.receipts.reduce((sum, r) => sum + r.total, 0);
  }

  onSold(listener: Listener): this {
    return this.on("sold", listener);
  }
}

export async function openTill(): Promise<Till> {
  const till = new Till();
  await Promise.resolve(null);
  return till;
}

File: Packages/CodeText/Tests/CodeTextTests/Samples/sample.swift
import Foundation

/// A small shop's till, for the colour samples (041).
///
/// Comments, strings, numbers, types and functions all appear at least once.
struct Till {
    enum Error: Swift.Error { case empty, overdrawn(by: Decimal) }

    private(set) var float: Decimal = 150.00
    var receipts: [Receipt] = []
    let opened = Date()

    /* A block comment
       across two lines. */
    mutating func sell(_ item: String, for price: Decimal, count: Int = 1) throws -> Receipt {
        guard count > 0 else { throw Error.empty }
        let total = price * Decimal(count)
        float += total
        let receipt = Receipt(item: item, total: total, at: .now)
        receipts.append(receipt)
        print("Sold \(count) × \(item) for \(total)")
        return receipt
    }

    mutating func refund(_ receipt: Receipt) throws {
        guard float >= receipt.total else {
            throw Error.overdrawn(by: receipt.total - float)
        }
        float -= receipt.total
        receipts.removeAll { $0.id == receipt.id }
    }

    var takings: Decimal {
        receipts.reduce(0) { $0 + $1.total }
    }
}

struct Receipt: Identifiable, Hashable {
    let id = UUID()
    let item: String
    let total: Decimal
    let at: Date
}

@MainActor
final class TillModel: ObservableObject {
    @Published private(set) var till = Till()
    private let hexMask = 0xFF_FF
    private let ratio = 0.125

    func sellCoffee() {
        do {
            _ = try till.sell("coffee", for: 3.20)
        } catch {
            print("Could not sell: \(error)")
        }
    }

    nonisolated static let greeting = """
        Welcome.
        Mind the step.
        """
}

let emoji = "Café ☕️ — naïve 👩‍👩‍👧 test"
